### PR TITLE
Adding new Energy Value formulation for Storage

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Adding new energy value formulation similar to water value in PLEXOS
* Missing Tests

### CODE REVIEW

File: src/devices_models/devices/storage.jl [hunk 1]
```diff
@@ -400,3 +416,14 @@ function objective_function!(
...
     return
 end
+
+function objective_function!(
+    container::OptimizationContainer,
+    devices::IS.FlattenIteratorWrapper{T},
+    ::DeviceModel{T, S},
+    ::Type{<:PM.AbstractPowerModel},
+) where {T <: PSY.Storage, S <: EnergyValue}
+    @show "called objective_function!"
```
Comment (daniel-thom) [inline src/devices_models/devices/storage.jl:426]: Debug print

**Review by claytonpbarrows — APPROVED** [review]
LGTM

**Review by claytonpbarrows — COMMENTED** [review]
LGTM

**Comment by jd-lara** [comment]
Closing since this code should move to https://github.com/NREL-SIIP/StorageSystemsSimulations.jl

**Pull request closed without merging**